languages/ruby/exercises/concept/exceptions/.docs/introduction.md

```diff
@@ -0,0 +1,73 @@
+Exceptions in Ruby as in most languages provide a way of dealing with unexpected events. Proper handling of exceptions is important when trying to prevent your program from crashing.
+
+When an exception is raised in your code, either by raising it explicitly or by Ruby interpreter raising it itself. The program stops and eventually exit with an error message:
+
+```ruby
+raise ArgumentError.new("Something went wrong!")
+=> Traceback (most recent call last):
+.
+.
+ArgumentError (Something went wrong!)
+```
+
+```ruby
+1/0
+=> Traceback (most recent call last):
+.
+.
+ZeroDivisionError (divided by 0)
+```
+
+In case we want to stop this shut down process we need to react to the error. This is called "rescuing" and exception:
+
+```ruby
+begin
+  # ...any code that raises an exception
+rescue
+  puts 'Got an exception'
+end
+```
+
+This program will not crash and it'll output "Got an exception". Instead of exiting, Ruby runs the code in the rescue block, which prints out a message.
+
+As everything in Ruby, exceptions are also objects and they usually hold data about the exception. This is how we can get the exception object:
+
+```ruby
+begin
+  # ...any code that raises an exception
+rescue => e
+  puts "Exception class: #{ e.class.name }"
+  puts"Exception Message:#{e.message}"
+end
+```
+
+In Ruby it's also possible to raise your own exceptions. For example:
+
+```ruby
+begin
+  raise ArgumentError.new("Invalid argument")
+rescue ArgumentError => e
+  puts e.message
+end
+```
+
+The previous exception is one of the [Ruby's built in exceptions][ruby-exceptions] but it's also possible to define custom exceptions and raise them:
+
+```ruby
+
+class CustomError < StandardError
+end
+
+raise CustomError.new()
+
+```
+
+```ruby
+UnsupportedOperation = Class.new(StandardError)
+
+raise UnsupportedOperation.new()
+end
+
```

Comment:
The example is the minimum.  Do we cover the signature of a custom `Exception` and the ability to provide for custom messaging?  Even a default message?

Reply:
Good point. I tried to keep the introduction to the minimum, but I agree, something about the `Exception` signature and one example will be nice to have. I'll add a small description an example

Reply:
>  I tried to keep the introduction to the minimum

Which is great!

> Do we cover the signature of a custom Exception and the ability to provide for custom messaging? Even a default message?

Remember to keep in mind that the introduction should only contain what is needed for the student to solve the exercise. If the exercise doesn't rely on custom message, it should not be part of the introduction (it can of course be part of the concept document).